### PR TITLE
update to new spec of extensions

### PR DESCRIPTION
**Pull request by templarfelix**
using method 

window.extensionsAPI.registerResourceExtension(

### CODE REVIEW

File: ui/webpack.config.js [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
   },
   output: {
-    filename: 'extensions.js',
-    path: __dirname + `/dist/resources/${groupKind}/ui`,
-    libraryTarget: 'window',
-    library: ['extensions', 'resources', groupKind],
+    filename: `extensions-${extName}.js`,
+    path: __dirname + `/dist/resources/extension-${extName}.js`,
+    libraryTarget: "window",
+    library: ["tmp", "extensions"],
```
Comment (ashutosh16) [inline ui/webpack.config.js:13]: Not sure what is usage here? Can you briefly explain

Comment (templarfelix) [reply ui/webpack.config.js:13]: I add the same id of Metrics Extension https://github.com/argoproj-labs/argocd-extension-metrics/blob/5d714b0b79453a3eb6c0315c8294e0a3fcb4fbdb/extensions/resource-metrics/resource-metrics-extention/ui/webpack.config.js#L16

File: manifests/install.yaml [hunk 2]
```diff
@@ -7,4 +7,4 @@ metadata:
 spec:
   sources:
   - web:
-      url: https://github.com/argoproj-labs/rollout-extension/releases/download/v0.3.1/extension.tar
+      url: https://github.com/argoproj-labs/rollout-extension/releases/download/v0.3.2/extension.tar
```
Comment (zachaller) [inline manifests/install.yaml:10]: Let's not bump this

Comment (templarfelix) [reply manifests/install.yaml]: reverted

**Review by zachaller — APPROVED** [review]

**Pull request merged**